pkg/server/updateResults.go

```diff
@@ -45,43 +48,87 @@ func updateResultRequest(ce *CreateExperiment) UpdateResults {
 	pm := kruize.NewProfileManager("")
 	for _, kubeobj := range ce.KubernetesObjects {
 		for _, contlist := range kubeobj.Containers {
-
-			// get queries from performanceProfile per container:
-			metricsList := pm.GetPerformanceProfileInstanceMetrics(ce.ClusterName, kubeobj.Namespace,
-				kubeobj.Name, contlist.ContainerName, ce.TrialSettings.MeasurementDuration)
-
 			//call function to parse the metrics:
 			windowSizeStr := strings.TrimSuffix(ce.TrialSettings.MeasurementDuration, "min")
 			windowSizeInt, _ := strconv.ParseInt(windowSizeStr, 10, 64)
-			startTime := time.Now()
-			endTime := startTime.Add(time.Duration(windowSizeInt) * time.Minute)
-			updateResult = UpdateResults{
-				Version:        ce.Version,
-				ExperimentName: ce.ExperimentName,
-				StartTimestamp: startTime.Format("2006-01-02T15:04:05.000Z"), //an hour ago from now
-				EndTimestamp:   endTime.Format("2006-01-02T15:04:05.000Z"),   //now
-				KubernetesObjects: []model.KubernetesObjectMetrics{
-					{
-						Type:      "deployment",
-						Name:      kubeobj.Name,
-						Namespace: kubeobj.Namespace,
-						Containers: []model.ContainerMetrics{
-							{
-								ContainerImage: contlist.ContainerImage,
-								ContainerName:  contlist.ContainerName,
-								Metrics:        metricsList,
+
+			endTimes := getTimeWindows(1, int(windowSizeInt))
+			for i := 1; i < len(endTimes); i++ {
+				startTime := endTimes[i-1]
+				endTime := endTimes[i]
+				unixTime := endTime.Unix()
+				// get queries from performanceProfile per container:
+				metricsList := pm.GetPerformanceProfileInstanceMetrics(ce.ClusterName, kubeobj.Namespace,
+					kubeobj.Name, contlist.ContainerName, ce.TrialSettings.MeasurementDuration, unixTime)
+
+				updateResult = UpdateResults{
+					Version:        ce.Version,
+					ExperimentName: ce.ExperimentName,
+					StartTimestamp: startTime.Format("2006-01-02T15:04:05.000Z"), //an hour ago from now
+					EndTimestamp:   endTime.Format("2006-01-02T15:04:05.000Z"),   //now
+					KubernetesObjects: []model.KubernetesObjectMetrics{
+						{
+							Type:      "deployment",
+							Name:      kubeobj.Name,
+							Namespace: kubeobj.Namespace,
+							Containers: []model.ContainerMetrics{
+								{
+									ContainerImage: contlist.ContainerImage,
+									ContainerName:  contlist.ContainerName,
+									Metrics:        metricsList,
+								},
 							},
 						},
 					},
-				},
+				}
+				updateResults = []UpdateResults{updateResult}
+				upJson, _ := json.Marshal(updateResults)
+				klog.V(9).Infof("Created updateResults Object %v", string(upJson))
```

Comment:
Arg is a string.
```suggestion
				klog.V(9).Infof("Created updateResults Object %s", string(upJson))
```

Or you can use `%+v` and pass the `upJson` object without converting to a string.